rust/worker/src/compactor/scheduler.rs

```diff
@@ -1,33 +1,42 @@
+use crate::assignment::assignment_policy::AssignmentPolicy;
 use crate::compactor::scheduler_policy::SchedulerPolicy;
 use crate::compactor::types::CompactionJob;
 use crate::log::log::CollectionInfo;
 use crate::log::log::CollectionRecord;
 use crate::log::log::Log;
+use crate::memberlist::Memberlist;
 use crate::sysdb::sysdb::SysDb;
 use uuid::Uuid;
 
-#[derive(Clone)]
 pub(crate) struct Scheduler {
+    my_ip: String,
     log: Box<dyn Log>,
     sysdb: Box<dyn SysDb>,
     policy: Box<dyn SchedulerPolicy>,
     job_queue: Vec<CompactionJob>,
     max_concurrent_jobs: usize,
```

Comment:
thinking: would it just be a cleaner pattern to have structs store their config object instead of unpacking it?